pkg/scalers/solace_scaler.go

```diff
@@ -0,0 +1,367 @@
+		}
+	} else if usernameClear, ok := config.TriggerMetadata[solaceMetaUsername]; ok && usernameClear != "" {
+		u = usernameClear
+	} else {
+		return "", "", fmt.Errorf("username is required and not found in K8Secret, environment, or clear text")
+	}
+	//	Get Password
+	if passwordSecret, ok := config.AuthParams[solaceMetaPassword]; ok && passwordSecret != "" {
+		p = passwordSecret
+	} else if passwordEnv, ok := config.TriggerMetadata[solaceMetaPasswordEnv]; ok && passwordEnv != "" {
+		if resolvedPassword, ok := config.ResolvedEnv[config.TriggerMetadata[solaceMetaPasswordEnv]]; ok && resolvedPassword != "" {
+			p = resolvedPassword
+		} else {
+			return "", "", fmt.Errorf("password could not be resolved from the environment variable: %s", passwordEnv)
+		}
+	} else if passwordClear, ok := config.TriggerMetadata[solaceMetaPassword]; ok && passwordClear != "" {
+		p = passwordClear
+	} else {
+		return "", "", fmt.Errorf("password is required and not found in K8Secret, environment, or clear text")
+	}
+	return u, p, nil
+}
+
+//	INTERFACE METHOD
+//	DEFINE METRIC FOR SCALING
+//	CURRENT SUPPORTED METRICS ARE:
+//	- QUEUE MESSAGE COUNT (msgCount)
+//	- QUEUE SPOOL USAGE   (msgSpoolUsage in MBytes)
+//	METRIC IDENTIFIER HAS THE SIGNATURE:
+//	- solace-[VPN_Name]-[Queue_Name]-[metric_type]
+//	e.g. solace-myvpn-QUEUE1-msgCount
+func (s *SolaceScaler) GetMetricSpecForScaling() []v2beta2.MetricSpec {
+	var metricSpecList []v2beta2.MetricSpec
+	// Message Count Target Spec
+	if s.metadata.msgCountTarget > 0 {
+		targetMetricValue := resource.NewQuantity(int64(s.metadata.msgCountTarget), resource.DecimalSI)
+		externalMetric := &v2beta2.ExternalMetricSource{
+			Metric: v2beta2.MetricIdentifier{
+				Name: kedautil.NormalizeString(fmt.Sprintf("%s-%s-%s-%s", solaceScalerID, s.metadata.messageVpn, s.metadata.queueName, solaceTriggermsgcount)),
+			},
+			Target: v2beta2.MetricTarget{
+				Type:         v2beta2.AverageValueMetricType,
+				AverageValue: targetMetricValue,
+			},
+		}
+		metricSpec := v2beta2.MetricSpec{External: externalMetric, Type: solaceExtMetricType}
+		metricSpecList = append(metricSpecList, metricSpec)
+	}
+	// Message Spool Usage Target Spec
+	if s.metadata.msgSpoolUsageTarget > 0 {
+		targetMetricValue := resource.NewQuantity(int64(s.metadata.msgSpoolUsageTarget), resource.DecimalSI)
+		externalMetric := &v2beta2.ExternalMetricSource{
+			Metric: v2beta2.MetricIdentifier{
+				Name: kedautil.NormalizeString(fmt.Sprintf("%s-%s-%s-%s", solaceScalerID, s.metadata.messageVpn, s.metadata.queueName, solaceTriggermsgspoolusage)),
+			},
+			Target: v2beta2.MetricTarget{
+				Type:         v2beta2.AverageValueMetricType,
+				AverageValue: targetMetricValue,
+			},
+		}
+		metricSpec := v2beta2.MetricSpec{External: externalMetric, Type: solaceExtMetricType}
+		metricSpecList = append(metricSpecList, metricSpec)
+	}
+	return metricSpecList
+}
```

Comment:
@coderanger do you think that the similar approcah would be beneficial for Rabbit MQ scaler?

Reply:
Do you need me to do anything with this or can I resolve?

Reply:
@dennis-brinley nope, this was just FYI for @coderanger :)